packages/devextreme/testing/tests/DevExpress.ui.widgets/gallery.markup.tests.js

```diff
@@ -78,6 +78,57 @@ QUnit.module('aria accessibility', () => {
         assert.equal($element.attr('aria-label'), 'gallery', 'widget should have aria-label to have difference from text list');
     });
 
+    QUnit.test('aria label for items', function(assert) {
+        const $gallery = $('#gallerySimple').dxGallery({ items: [1] });
+
+        const $item = $gallery.find(`.${GALLERY_ITEM_CLASS}`);
+
+        assert.equal($item.attr('aria-label'), 'gallery item', 'item should have "aria-label"');
+    });
+
+    QUnit.test('aria-activedescendant should have link to one element only', function(assert) {
+        const $gallery = $('#gallerySimple').dxGallery({
+            height: 100,
+            width: '100%',
+            showIndicator: false,
+            items: [1, 2],
+            loop: true
+        });
+
+        const $items = $gallery.find(`.${GALLERY_ITEM_CLASS}`);
+
+        const itemsWithId = Array.from($items).filter(item => {
+            return Boolean($(item).attr('id'));
+        });
+
+        if(windowUtils.hasWindow()) {
+            assert.strictEqual(itemsWithId.length, 1, 'id attribute should be exist only on one item');
+        } else {
+            assert.strictEqual(itemsWithId.length, 0, 'no rendered elements');
+        }
+    });
+
+    QUnit.test('aria-activedescendant should have link to one element only when width was not set', function(assert) {
```

Comment:
[unclear test]
but test above shows that activedescendant is set when width is set to `100%`.
I didn't catch how should it work.
If you wanna to test that activedescendant element points to the single elements in both cases when width is specified or nor specified, let's write a single test in `forEach`

Reply:
fixed